controllers/nodehealthcheck_controller.go

```diff
@@ -106,32 +105,69 @@ func (r *NodeHealthCheckReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		}
 	}()
 
+	// check if we need to disable NHC because of invalid configuration
+	// Remove this and corresponding test when kubebuilder supports minimum on IntOrStr types
+	if err = utils.ValidateMinHealthy(nhc); err != nil {
+		// update status if needed
+		if !utils.IsConditionTrue(nhc.Status.Conditions, remediationv1alpha1.ConditionTypeDisabled, remediationv1alpha1.ConditionReasonDisabledInvalidConfig) {
+			log.Info("disabling NHC because of invalid config")
+			meta.SetStatusCondition(&nhc.Status.Conditions, metav1.Condition{
+				Type:    remediationv1alpha1.ConditionTypeDisabled,
+				Status:  metav1.ConditionTrue,
+				Reason:  remediationv1alpha1.ConditionReasonDisabledInvalidConfig,
+				Message: err.Error(),
+			})
+			r.Recorder.Eventf(nhc, eventTypeWarning, eventReasonDisabled, "Invalid configuration: %s", err.Error())
+		}
+		// stop reconciling
+		return result, nil
+	}
+
 	// check if we need to disable NHC because of existing MHCs
 	if disable := r.MHCChecker.NeedDisableNHC(); disable {
 		// update status if needed
-		if !meta.IsStatusConditionTrue(nhc.Status.Conditions, remediationv1alpha1.ConditionTypeDisabled) {
+		if !utils.IsConditionTrue(nhc.Status.Conditions, remediationv1alpha1.ConditionTypeDisabled, remediationv1alpha1.ConditionReasonDisabledMHC) {
```

Comment:
ASFAI we're always using `!`utils.IsConditionTrue.
Perhaps change the implementation of IsConditionTrue so we can get rid of the `!`

Reply:
hm, I think `!isTrue` is exactly what we want to check. Do you think `isFalseOrUnknown` is better?
Also, at least in theory, there might be even more potential status values (uncommon, but not impossibe, it's a string and not a bool)

Reply:
>hm, I think !isTrue is exactly what we want to check. Do you think isFalseOrUnknown is better?

Yes I do.
I think readability wise most people will wrongly assume that  !isTrue == isFalse.
`isFalseOrUnknown` serves two purposes:
1. getting rid of this ambiguity
2. simplifying the code by removing the `!` operator

Reply:
but still, it's a string which can be anything, so isFalseOrUnknown doesn't work 🤷🏼‍♂️